crates/ika-core/src/dwallet_mpc/mod.rs

```diff
@@ -406,13 +408,16 @@ pub(crate) fn advance_and_serialize<P: AsynchronouslyAdvanceable>(
         Err(e) => {
             let general_error = DwalletMPCError::TwoPCMPCError(format!(
                 "MPC error in party {party_id} session {} at round #{} {:?}",
-                session_id,
-                messages.len() + 1,
-                e
+                session_id, mpc_round, e
             ));
             return match e.into() {
                 // No threshold was reached, so we can't proceed.
                 mpc::Error::ThresholdNotReached { honest_subset } => {
+                    let round_to_restart = P::round_to_retry_on_threshold_not_reached(mpc_round)
+                        .expect(&format!(
+                            "failed to get the round to retry from, this should never happen. {mpc_round} {session_id}",
+                        ));
```

Comment:
1. Print the error with all the possible context, use error!() macro
2. Do we want the Validator to die on this case? @ycscaly Because `expect()` kills it.